doc/source/tune/faq.rst

```diff
@@ -280,6 +280,33 @@ on other nodes as well. Please refer to the
 :ref:`placement groups documentation <ray-placement-group-doc-ref>` to learn more
 about these placement strategies.
 
+You can also allocate specific resources to a trial based on a custom rule via lambda functions.
+For instance, if you want to allocate GPU resources to trials based on a setting in your config:
+
+.. code-block:: python
+
+    tuner = tune.Tuner(
```

Comment:
@heyitsmui can you please make this a testable `literalinclude` block? That should be easy to convert, you can have a look at various examples in the same file.

Reply:
thanks @maxpumperla! will add

Reply:
@maxpumperla just fixed, please take a look when you have a moment, thanks!

Reply:
@heyitsmui the linter fails now, other than that we should be good here. Fails in two places:

```
tuner = tune.Tuner(
--
  | tune.with_resources(
  | train_fn,
  | -            resources=lambda spec: {"GPU": 1} if config.use_gpu else {"GPU": 0})
  | +            resources=lambda spec: {"GPU": 1} if config.use_gpu else {"GPU": 0},
  | +        )
  | )
  | tuner.fit()
  | # __resources_lambda_end__
  | @@ -107,9 +108,11 @@ if not MOCK:
  | train_fn,
  | param_space={
  | "scaling_config": ScalingConfig(
  | -                trainer_resources=lambda spec: {"GPU": 1} if config.use_gpu else {"GPU": 0}
  | -                )
  | -        }
  | +                trainer_resources=lambda spec: {"GPU": 1}
  | +                if config.use_gpu
  | +                else {"GPU": 0}
  | +            )
  | +        },
  | )
```

Reply:
thanks! just put out a fix, will wait for all checks to pass first

Reply:
thanks again @maxpumperla all tests have passed now!